sbin/umount/umount.c

```diff
@@ -455,6 +469,16 @@ umountfs(struct statfs *sfs)
 		clnt_destroy(clp);
 	}
 	free(orignfsdirname);
+
+	if (dflag) {
+		if (md_detach(sfs->f_mntfromname) == 0) {
+			if (vflag)
+				(void)printf("%s: detached\n",
+				    sfs->f_mntfromname);
+		} else if (!all)
+			return (-1);
```

Comment:
so this means umount will return a failure code if -d is specified on non /dev/md backed devices. Is this what you want?

Reply:
> so this means umount will return a failure code if -d is specified on non /dev/md backed devices. Is this what you want?

Yes, but only if the `-a` wasn't specified for obvious reasons.